src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            head.next = new Node<>(head, value, null);
+            head = head.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index: " + index + " for size: " + size);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            tail = new Node<>(null, value, tail);
+        } else {
+            Node<T> nextNode = getNode(index);
+            nextNode.prev.next = new Node<>(nextNode.prev, value, nextNode);
+            nextNode.prev = nextNode.prev.next;
+        }
+        size++;
```

Comment:
then reorder code blocks to simplify code and reuse existing method
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndexValidation(index);
        if (index == 0) {
            tail = new Node<>(null, value, tail);
        } else {
            Node<T> nextNode = getNode(index);
            nextNode.prev.next = new Node<>(nextNode.prev, value, nextNode);
            nextNode.prev = nextNode.prev.next;
        }
        size++;
```